pallets/committee-management/src/manager.rs

```diff
@@ -0,0 +1,143 @@
+/// * Session ends:
+/// 1. Inner T::end_session invoked
+/// 2. Logic related to new session from this pallet is invoked
+/// * Session starts:
+/// 1. Inner T::start_session invoked
+/// 2. Logic related to new session from this pallet is invoked
+/// 3. If session starts era EM::new_era_start invoked
+/// 4. If session starts era logic related to new era from this pallet is invoked
+///
+/// In the runtime we set EM to pallet_elections and T to combination of staking and historical_session.
+pub struct SessionAndEraManager<E, EM, T, C>(PhantomData<(E, EM, T, C)>)
+where
+    E: EraInfoProvider,
+    EM: EraManager,
+    T: SessionManager<C::AccountId>,
+    C: Config;
+
+impl<E, EM, T, C> SessionAndEraManager<E, EM, T, C>
+where
+    E: EraInfoProvider,
+    EM: EraManager,
+    T: SessionManager<C::AccountId>,
+    C: Config,
+{
+    fn session_starts_era(session: SessionIndex) -> Option<EraIndex> {
+        let active_era = match E::active_era() {
+            Some(ae) => ae,
+            // no active era, session can't start it
+            _ => return None,
+        };
+
+        if Self::is_start_of_the_era(active_era, session) {
+            return Some(active_era);
+        }
+
+        None
+    }
+
+    fn session_starts_next_era(session: SessionIndex) -> Option<EraIndex> {
+        let active_era = match E::active_era() {
+            Some(ae) => ae + 1,
+            // no active era, session can't start it
+            _ => return None,
+        };
+
+        if Self::is_start_of_the_era(active_era, session) {
+            return Some(active_era);
+        }
+
+        None
+    }
+
+    fn is_start_of_the_era(era: EraIndex, session: SessionIndex) -> bool {
+        if let Some(era_start_index) = E::era_start_session_index(era) {
+            return era_start_index == session;
+        }
+
+        false
+    }
+}
+
+impl<E, EM, T, C> SessionManager<C::AccountId> for SessionAndEraManager<E, EM, T, C>
+where
+    E: EraInfoProvider,
+    EM: EraManager,
+    T: SessionManager<C::AccountId>,
+    C: Config,
+{
+    fn new_session(new_index: SessionIndex) -> Option<Vec<C::AccountId>> {
+        T::new_session(new_index);
+        if let Some(era) = Self::session_starts_next_era(new_index) {
+            EM::on_new_era(era);
+            Pallet::<C>::emit_fresh_bans_event();
+        }
+
+        Pallet::<C>::rotate_committee(new_index)
+    }
+
+    fn end_session(end_index: SessionIndex) {
+        T::end_session(end_index);
+        Pallet::<C>::adjust_rewards_for_session();
+        Pallet::<C>::calculate_underperforming_validators();
+        // clear block count after calculating stats for underperforming validators, as they use
+        // SessionValidatorBlockCount for that
+        let result = SessionValidatorBlockCount::<C>::clear(u32::MAX, None);
+        debug!(target: "pallet_elections", "Result of clearing the `SessionValidatorBlockCount`, {:?}", result.deconstruct());
```

Comment:
```suggestion
        debug!(target: "pallet_committee_management", "Result of clearing the `SessionValidatorBlockCount`, {:?}", result.deconstruct());
```

Reply:
If you could perhaps do the thing with `LOG_TARGET` that we have in some of our modules, so it's easier to stay consistent.